theano/compile/tests/test_builders.py

```diff
@@ -107,6 +107,19 @@ def test_shared_grad(self):
         fn = function([x, y, z], f)
         assert numpy.allclose(15.0 + s.get_value(),
                               fn(xv, yv, zv))
+    
+    def test_connection_pattern(self):
+        import numpy
+        x, y, z = T.matrices('xyz')
+        out1 = x * y
+        out2 = y * z
+
+        op = OpFromGraph([x ,y, z], [out1, out2], moe='FAST_RUN')
+        results = op.connection_pattern(None)
+        expect_result = [[True, False],
+                         [True, True],
+                         [False, True]]
+        assert results == expect_result
```

Comment:
This test is very basic. To offer better coverage, there are additional cases it should test, including; 
- Inner graph with ops that don't have a 'full' connection pattern (meaning not every output is connected to every input)
- Inner graph with ops that have multiple outputs (a nested OpFromGraph, for instance)
- Inner graph where some computation doesn't rely on explicit inputs to the OpFromGraph (constants and random number sampling)